pennylane/interfaces/torch.py

```diff
@@ -246,3 +258,289 @@ def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_d
         max_diff=max_diff,
     )
     return ExecuteTapes.apply(kwargs, *parameters)
+
+
+def pytreeify(cls):
+    """Pytree class"""
+    orig_fw = cls.forward
+    orig_bw = cls.backward
+    orig_apply = cls.apply
+
+    def new_apply(*inp):
+        # Inputs already flat
+        out_struct_holder = []
+        flat_out = orig_apply(out_struct_holder, *inp)
+        return pytree.tree_unflatten(flat_out, out_struct_holder[0])
+
+    def new_forward(ctx, out_struct_holder, *inp):
+        out = orig_fw(ctx, *inp)
+        flat_out, out_struct = pytree.tree_flatten(out)
+        ctx._out_struct = out_struct
+        out_struct_holder.append(out_struct)
+        return tuple(flat_out)
+
+    def new_backward(ctx, *flat_grad_outputs):
+        grad_outputs = pytree.tree_unflatten(flat_grad_outputs, ctx._out_struct)
+        grad_inputs = orig_bw(ctx, *grad_outputs)
+        return (None,) + tuple(grad_inputs)

```

Comment:
Does this `None` gradient refer to the `out_struct_holder` argument of `new_forward`? In any case, can we add a comment here like line 449-450?

Reply:
Completely right I will do that.